ui/src/components/SampleView/MotorControls.jsx

```diff
@@ -1,113 +1,69 @@
-import React from 'react';
-import { Button, Row, Col } from 'react-bootstrap';
-import MotorInputContainer from '../../containers/MotorInputContainer';
+import React, { useState } from 'react';
+import { Button } from 'react-bootstrap';
+import { useSelector } from 'react-redux';
 
+import MotorInputContainer from '../../containers/MotorInputContainer';
 import TwoAxisTranslationControl from '../MotorInput/TwoAxisTranslationControl';
-import { find } from 'lodash';
+import styles from './MotorControls.module.css';
 
 import '../MotorInput/motor.css';
 
-export default class MotorControls extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = { showAll: false };
-  }
-
-  renderMotorInputs(from, to) {
-    return this.props.uiproperties
-      .slice(from, to)
-      .map(({ attribute, role }) => (
-        <Col key={attribute} sm={12}>
-          <MotorInputContainer component="sample_view" role={role} />
-        </Col>
-      ));
-  }
-
-  render() {
-    const sample_vertical_uiprop = find(this.props.uiproperties, {
-      role: 'sample_vertical',
-    });
+function MotorControls() {
+  const [showAll, setShowAll] = useState(false);
 
-    const sample_horizontal_uiprop = find(this.props.uiproperties, {
-      role: 'sample_horizontal',
-    });
+  const motorsProps = useSelector((state) =>
+    state.uiproperties.sample_view.components.filter(
+      ({ value_type }) => value_type === 'MOTOR',
+    ),
+  );
 
-    const sample_vertical = find(this.props.hardwareObjects, {
-      name: sample_vertical_uiprop.attribute,
-    });
+  const verticalMotorProps = motorsProps.find(
+    (c) => c.role === 'sample_vertical',
+  );
+  const horizontalMotorProps = motorsProps.find(
+    (c) => c.role === 'sample_horizontal',
+  );
 
-    const sample_horizontal = find(this.props.hardwareObjects, {
-      name: sample_horizontal_uiprop.attribute,
-    });
-
-    const numel = this.props.uiproperties.length;
-
-    if (!sample_vertical || !sample_horizontal) {
-      return <Row className="row">{this.renderMotorInputs(0, numel)}</Row>;
-    }
+  if (!verticalMotorProps || !horizontalMotorProps) {
+    return motorsProps.map(({ attribute: k, role }) => (
+      <MotorInputContainer key={k} component="sample_view" role={role} />
```

Comment:
Note that I haven't refactored `MotorInputContainer` and `MotorInput`, so things might evolve further.